### PR TITLE
Blog post about flink native integration

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**

### CODE REVIEW

**Review by arturowczarek — APPROVED** [review]

File: website/blog/flink-native-support/index.mdx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+### Towards native lineage support in Flink
+
+The major complexity of implementing this is that it requires writing code in separate repositories with different release cadence.
+
+The first step taken was to create lineage interfaces’ definitions in the Flink core repository, which was released in Flink 1.19.
+Flink core has also been extended with a mechanism to create lineage graphs out of the sinks’ and sources’ lineage interfaces as well as expose the graph through the job listener.
+This has already been merged to Flink’s main branch and will be released in Flink 2.0.
+
+Next step is to implement lineage for the sources and sinks which oftentimes in separate repositories.
```
Comment (JDarDagran) [inline website/blog/flink-native-support/index.mdx:50]: The sentence is not grammatically correct? I think word _are_ is missing.

File: website/blog/flink-native-support/index.mdx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+Apache Flink is a powerful stream processing engine widely adopted in the industry.
+Flink jobs are capable of emitting OpenLineage events for already two years.
+However, it is a recent joint effort of Flink and OpenLineage communities that will bring this integration into the new era.
+
+<!--truncate-->
+
+### Existing OpenLineage Flink integration
+
+As of November 2024, [OpenLineage Flink integration](https://github.com/OpenLineage/OpenLineage/tree/main/integration/flink) supports Flink versions `1.15` to `1.18` and emits OpenLineage events complying with the dataset naming model.
+It supports Kafka sources and sinks with Avro and Protobuf, JDBC and Cassandra connectors with extracting lineage from the queries executed.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:16]: quite a lot of `and`'s and `with`'s and commas for one sentence

I don't have good suggestion on how to rephrase that yet

File: website/blog/flink-native-support/index.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+It also supports Hybrid sources. Overall, this is extremely helpful for the organizations to collect consistent cross-platform lineage graphs.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:17]: `this is extremly` -> `it is extremly`

Comment (tnazarew) [reply website/blog/flink-native-support/index.mdx:17]: also for now it seems to me to say that "In general its helpful for the organisation to have have cross platform lineage" and I think you wan to say that "these features can aid organisations greatly in lineage collection"

File: website/blog/flink-native-support/index.mdx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+The integration has been implemented with no changes to the Flink codebase which allowed OpenLineage community to boost its implementation in the early phase and recognize some pain points to be solved later.
+
+First, it is based on reflection mechanisms in Java and oftentimes extracts class members in an unsupported way.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:21]: what does unsupported way mean?

File: website/blog/flink-native-support/index.mdx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+This makes the package difficult to maintain, causing from time to time classloading issues.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:22]: either `causing, from time to time, classloading issues.` or `causing classloading issues from time to time.`

File: website/blog/flink-native-support/index.mdx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+The other functional limitations are: it does not support Flink SQL and it requires a job's code change to initialize the OpenLineage listener.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:24]: `are: not supporting Flink SQL and it requiring a job's`

File: website/blog/flink-native-support/index.mdx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+Although only a tiny code modification is required, it may be a blocker for turning OpenLineage on immediately for all the Flink jobs within the organisation.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:25]: `for enabling OpenLineage on all Flink jobs` maybe... not sure here

File: website/blog/flink-native-support/index.mdx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+### Flip Pi 3.1415… to the rescue
+
+[Flip-314](https://cwiki.apache.org/confluence/display/FLINK/FLIP-314%3A+Support+Customized+Job+Lineage+Listener) is Flink’s community proposal to let Flink jobs expose lineage metadata via official interfaces.
+The overall idea is to let Flink’s source and sinks classes expose metadata about datasets consumed and produced.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:30]: `sources and sinks`

Comment (tnazarew) [reply website/blog/flink-native-support/index.mdx:30]: `consumed and produced datasets`

File: website/blog/flink-native-support/index.mdx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+![Code diagram](./diagram.png)
+
+This is where we are now. The next implementation steps are:
+
+ * Implement lineage interfaces for other Flink connectors.
+ * Release new OpenLineage listener.
+
+We also need to wait for the Flink community to release Flink 2.0.
+Further work is required to design and implement column level lineage for Flink SQL.
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:70]: for the sake of consistency I would go for 

 * Implement lineage interfaces for other Flink connectors.
 * Release new OpenLineage listener.
 * Wait for the Flink community to release Flink 2.0.
 * Design and implement column level lineage for Flink SQL.
 
 or at lease not mix passive and active in last two

File: website/blog/flink-native-support/index.mdx [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+### Summary
+
+The Flink community is going to have one of the best lineage support across the industry.
+As a native implementation, it guarantees there is no mismatch between data processing code and lineage extraction logic,
+as they both belong to the same codebase.
+Source and sink classes define input and output datasets for the data processing job,
```
Comment (tnazarew) [inline website/blog/flink-native-support/index.mdx:77]: `Sources' and sink's`

**Review by tnazarew — APPROVED** [review]

**Pull request merged**